src/Doctrine/ManagerRegistry.php

```diff
@@ -29,7 +29,12 @@ class ManagerRegistry implements ManagerRegistryInterface
     /**
      * @var EntityManager[]
      */
-    protected $managers;
+    protected $originalManagers;
+
+    /**
+     * @var EntityManager[]
+     */
+    protected $resetManagers = [];
```

Comment:
Hello @ghola 

Thank you for your pull request.  And the passion.

The only thing that i do not understand why you need originalManagers and resetManagers. I looks more complex to me as it used to.

Regards @dominikzogg